Sources/SwiftProtobuf/Internal.swift

```diff
@@ -40,7 +40,7 @@ public enum Internal {
 
   /// Helper to loop over dictionary with values that are Messages to see if
   /// they are all initialized (see Message.isInitialized for what that means).
-  public static func areAllInitialized<K: Hashable>(_ mapToMessages: [K: Message]) -> Bool {
+  public static func areAllInitialized<K>(_ mapToMessages: [K: Message]) -> Bool {
```

Comment:
Yep.  This works in Swift Linux 3.0.1 and later.  I was a little surprised as well; the type-checker is clearly smarter than I gave it credit for.

I think this should work for Xcode 8.2; I'll track that down and try it.